viewing_party/party.py

```diff
@@ -0,0 +1,130 @@
+#wave 1
+def create_movie(movie_title,genre,rating):
+    create_movie_dict = {'title': movie_title, 'genre': genre, 'rating':rating}
+    
+
+    if movie_title and genre and rating:
+        movie_dict = {'title':movie_title, 'genre':genre, 'rating':rating}
+        return movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data,movie_title):
+    watched_list = user_data['watched']
+    watched_list.append(movie_title)
+    return user_data
+
+def add_to_watchlist(user_data,movie_title):
+    add_watchlist = user_data['watchlist']
+    add_watchlist.append(movie_title)
+    return user_data
+
+def watch_movie(user_data,movie_title):
+    movies_watched = user_data['watched']
+    movies_to_watch = user_data['watchlist']
+
+    for movie in movies_to_watch:
+        if movie_title == movie['title']:
+            movies_watched.append(movie)
+            movies_to_watch.remove(movie)
+
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+    # avg_rating = 0.0
+    # rating_sum = 0
+    # for i in range (len(user_data["watched"])):
+    #     rating_sum += user_data["watched"][i]["rating"]
+    #     #calculating
+    # rating_sum/len(user_data["watched"])
+    total_rating = 0
+    if len(user_data["watched"]) == 0:
+        return 0
+    else:
+        for movie in user_data["watched"]:
+            total_rating += movie["rating"]
+    return total_rating/len(user_data["watched"])
+def get_most_watched_genre(user_data):
+    most_recurring_genre= {}
+    if len(user_data["watched"]) == 0: 
+        return None 
+    for movie_dict in user_data["watched"]:
+        if movie_dict["genre"] in most_recurring_genre:
+            most_recurring_genre[movie_dict["genre"]] +=1
+        else:
+            most_recurring_genre[movie_dict["genre"]] = 1
+    max_genre = max(most_recurring_genre, key=most_recurring_genre.get)
+    return max_genre
+# Wave 3
+def get_unique_watched(user_data):
+    friends_list = []
+    user_only_watched = []
+    for watched_list in user_data["friends"]:
+        for movie in watched_list["watched"]:
+            friends_list.append(movie["title"])
+    for movie in user_data["watched"]:
+        if movie["title"] not in friends_list:
+            user_only_watched.append(movie) 
+    return user_only_watched
+def get_friends_unique_watched(user_data):
+    movies_friends_watched = []
+    for watched_list in user_data["friends"]:
+        for movie in watched_list["watched"]:
```

Comment:
this is a bit of repeat code, something you also had to do in `get_unique_watched`. This is a perfect signal to pull these for loops out and turn them into their own standalone help functions that create lists for you that a function can then manipulate!